### PR TITLE
Update Haskell package set to Stackage Nightly 2021-02-02 (plus other fixes)

### PR DESCRIPTION
**Pull request by peti**
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on Friday, 2021-02-05 20:00 +01:00. You can watch this live on Twitch at https://www.twitch.tv/peti343.

## TODO

- [x] https://github.com/NixOS/docker/pull/31: Use a pure Nix approach to building containers

### CODE REVIEW

File: pkgs/development/compilers/ghc/8.10.3.nix [hunk 1]
```diff
@@ -107,7 +107,15 @@ stdenv.mkDerivation (rec {
...
-  patches = lib.optionals stdenv.isDarwin [
+  patches = [
+    # See upstream patch at
+    # https://gitlab.haskell.org/ghc/ghc/-/merge_requests/4885. Since we build
+    # from source distributions, the auto-generated configure script needs to be
+    # patched as well, therefore we use an in-tree patch instead of pulling the
+    # upstream patch. Don't forget to check backport status of the upstream patch
+    # when adding new GHC releases in nixpkgs.
+    ./respect-ar-path.patch
+  ] ++ stdenv.lib.optionals stdenv.isDarwin [
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/ghc/8.10.3.nix:118]:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by maralorn** [comment]
@peti Have you already opened a new PR?

**Comment by peti** [comment]
Now I have. Tonight, we'll merge https://github.com/NixOS/nixpkgs/pull/112893.

**Pull request merged**